### PR TITLE
PYTHON-4219 Prep for 4.6.2 Release

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

File: pymongo/_version.py [hunk 1]
```diff
@@ -17,7 +17,7 @@
 
 from typing import Tuple, Union
 
-version_tuple: Tuple[Union[int, str], ...] = (4, 6, 2, ".dev0")
+version_tuple: Tuple[Union[int, str], ...] = (4, 6, 2, "")
```
Comment (ShaneHarvey) [inline pymongo/_version.py:20]: Historically we always remove the "".

File: doc/changelog.rst [hunk 2]
```diff
@@ -1,6 +1,13 @@
 Changelog
 =========
 
+Changes in Version 4.6.2
+------------------------
+
+PyMongo 4.6.2 fixes the following bug:
+
+- Python 3.12 thread error while using ``MongoClient()`` function.
```
Comment (ShaneHarvey) [inline doc/changelog.rst:9]: Could you use the same changelog entry that's in 4.7:
```
- Fixed a bug appearing in Python 3.12 where "RuntimeError: can't create new thread at interpreter shutdown"
  could be written to stderr when a MongoClient's thread starts as the python interpreter is shutting down.
```

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]
We still need to backport the fix for PYTHON-4147.

File: doc/changelog.rst (hunk 2 above)
Comment (blink1073) [reply doc/changelog.rst:9]: Done

**Comment by blink1073** [comment]
> We still need to backport the fix for [PYTHON-4147](https://jira.mongodb.org/browse/PYTHON-4147).

https://github.com/mongodb/mongo-python-driver/pull/1532

**Review by ShaneHarvey — APPROVED** [review]

**Review by caseyclements — APPROVED** [review]
lgtm

**Pull request merged**